test/metatx/ERC2771Forwarder.test.js

```diff
@@ -208,21 +216,30 @@ contract('ERC2771Forwarder', function (accounts) {
     });
 
     it('bubbles out of gas', async function () {
-      const receiver = await CallReceiverMock.new();
-      const gasAvailable = 100000;
-      this.requestData.to = receiver.address;
-      this.requestData.data = receiver.contract.methods.mockFunctionOutOfGas().encodeABI();
-      this.requestData.gas = 1000000;
-
+      this.requestData.data = this.receiver.contract.methods.mockFunctionOutOfGas().encodeABI();
+      this.requestData.gas = 1_000_000;
       this.requestData.signature = this.sign(this.alice.getPrivateKey());
 
+      const gasAvailable = 100_000;
       await expectRevert.assertion(this.forwarder.execute(this.requestData, { gas: gasAvailable }));
 
       const { transactions } = await web3.eth.getBlock('latest');
       const { gasUsed } = await web3.eth.getTransactionReceipt(transactions[0]);
 
       expect(gasUsed).to.be.equal(gasAvailable);
     });
+
+    it('bubbles out of gas forced by the relayer', async function () {
+      // A malicious relayer would provide a gas limit that is too low to execute the request.
+      const estimate = await this.estimateRequest(this.request);
+      const gasAvailable = estimate + 2_000; // Add some gas to let the first-level transaction pass
```

Comment:
I'm worried this value (2_000) and its equivalent on the batch version (10_000) is arbitrary because if there's not enough gas, the transaction won't be even sent because it will revert with a "not enough gas" error before sending.

I don't have an idea of how to justify this value better.

Reply:
Could we perhaps estimate the simplest request possible and use that value as the offset?

Reply:
The current `this.request` is a very simple request already, its estimation even adding the 21000 buffer that won't go in the subcall but covers initiating a relayed request:

```js
this.estimateRequest = request =>
      web3.eth.estimateGas({
        from: this.forwarder.address,
        to: request.to,
        data: web3.utils.encodePacked({ value: request.data, type: 'bytes' }, { value: request.from, type: 'address' }),
        value: request.value,
        gas: request.gas,
      });
```

In order to provide an accurate buffer we have to calculate how much gas is consumed from starting the transaction to the request's `CALL` opcode and use it as buffer. Also, doing an arbitrarily high buffer will make the underlying transaction not to revert.

We agreed internally to keep both so I'm solving